lib/Conversion/LTLToCore/LTLToCore.cpp

```diff
@@ -0,0 +1,237 @@
+          // Generate the sv assertion using the input to the parenting
+          // clock
+          rewriter.create<sv::AssertOp>(
+              loc, adaptor.getInput(),
+              sv::DeferAssertAttr::get(getContext(),
+                                       sv::DeferAssert::Immediate),
+              assertOp.getLabelAttr());
+        });
+
+    return success();
+  }
+};
+
+} // namespace
+
+//===----------------------------------------------------------------------===//
+// Lower LTL To Core pass
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct LowerLTLToCorePass : public LowerLTLToCoreBase<LowerLTLToCorePass> {
+  LowerLTLToCorePass() = default;
+  void runOnOperation() override;
+};
+} // namespace
+
+// Applies all of the conversion patterns needed for this lowering
+void circt::populateLTLToCoreConversionPatterns(
+    circt::LTLToCoreTypeConverter &converter,
+    mlir::RewritePatternSet &patterns) {
+  patterns.add<HasBeenResetOpConversion, LowerClockRelatedOpPatterns,
+               DisableOpConversion>(converter, patterns.getContext());
+}
+
+// Creates the type conversions and materializations needed for this pass to
+// work
+circt::LTLToCoreTypeConverter::LTLToCoreTypeConverter() {
+
+  // Convert the ltl property type to a built-in type
+  addConversion([](Type type) { return type; });
+  addConversion([](ltl::PropertyType type) {
+    return IntegerType::get(type.getContext(), 1);
+  });
+
+  // Basic materializations
+  addTargetMaterialization(
+      [&](mlir::OpBuilder &builder, mlir::Type resultType,
+          mlir::ValueRange inputs,
+          mlir::Location loc) -> std::optional<mlir::Value> {
+        if (inputs.size() != 1)
+          return std::nullopt;
+        return inputs[0];
+      });
+
+  addSourceMaterialization(
+      [&](mlir::OpBuilder &builder, mlir::Type resultType,
+          mlir::ValueRange inputs,
+          mlir::Location loc) -> std::optional<mlir::Value> {
+        if (inputs.size() != 1)
+          return std::nullopt;
+        return inputs[0];
+      });
+}
+
+// Simply applies the conversion patterns defined above
+void LowerLTLToCorePass::runOnOperation() {
+
+  // Set target dialects: We don't want to see any ltl or verif that might come
+  // from an AssertProperty left in the result
+  ConversionTarget target(getContext());
+  target.addLegalDialect<hw::HWDialect>();
+  target.addLegalOp<hw::ConstantOp>();
+  target.addLegalDialect<comb::CombDialect>();
+  target.addLegalOp<comb::OrOp>();
+  target.addLegalOp<comb::MuxOp>();
+  target.addLegalDialect<sv::SVDialect>();
+  target.addLegalOp<sv::AssertOp>();
+  target.addLegalOp<sv::AlwaysOp>();
+  target.addLegalDialect<seq::SeqDialect>();
+  target.addLegalOp<seq::CompRegOp>();
+  target.addIllegalOp<verif::HasBeenResetOp>();
+  target.addIllegalOp<verif::AssertOp>();
+  target.addIllegalOp<ltl::DisableOp>();
```

Comment:
Yes I removed `ltl::ClockOp` while debugging, I'll add it back. The entire dialects should not be made illegal as this conversion only cares about the parts used to define the `assert property` statement. I will make the entire dialects illegal when there exists a more complete lowering. If you really insist I can do it now, but it would make this pass unusable on some perfectly legal bits of `mlir`